docroot/modules/custom/mass_microsites/mass_microsites.module

```diff
@@ -146,3 +186,84 @@ function mass_microsites_entity_hierarchy_microsite_links_alter(array &$links) {
     }
   }
 }
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function mass_microsites_theme_suggestions_block_alter(array &$suggestions, array &$variables) {
+  if (_mass_microsites_get_microsite()) {
+    if ($variables['elements']['#plugin_id'] === 'mass_microsite_menu') {
+      if ($variables['elements']['#id'] === 'mass_theme_nearestmicrositemenu') {
+        $suggestions[] = 'block__mass_microsite_desktop_nav';
+      }
+
+      if ($variables['elements']['#id'] === 'mass_theme_nearestmicrositemenu_2') {
+        $suggestions[] = 'block__mass_microsite_header_bar';
+      }
+    }
+  }
+}
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ */
+function mass_microsites_theme_suggestions_page_alter(array &$suggestions, array &$variables) {
+  $node = _mass_microsites_get_node();
+
+  if ($node && _mass_microsites_get_microsite()) {
+
+    $without_main = [
+      'binder',
+      'info_details',
+      'guide_page',
+      'service_page',
+    ];
+    if (in_array($node->getType(), $without_main)) {
+      array_splice($suggestions, 1, 0, 'page__microsite__node__without_main');
+    }
+    else {
+      array_splice($suggestions, 1, 0, 'page__microsite');
+    }
+  }
+}
+
+/**
+ * Helper function to get the nearest microsite.
+ *
+ * @return \Drupal\entity_hierarchy_microsite\Entity\MicrositeInterface|False
+ *   The microsite object or FALSE if no microsite is found.
+ */
+function _mass_microsites_get_microsite() {
+  if ($node = _mass_microsites_get_node()) {
+    /** @var Drupal\entity_hierarchy_microsite\Entity\MicrositeInterface */
+    $microsite = \Drupal::service('mass_microsites.nearest_microsite_lookup')->getNearestMicrosite($node);
+    if ($microsite) {
+      return $microsite;
+    }
+  }
+
+  return FALSE;
+}
+
+/**
+ * Helper function to get the node based on the current route.
+ *
+ * @return \Drupal\node\NodeInterface|False
+ *   The node object or FALSE if no node is found.
+ */
+function _mass_microsites_get_node() {
+  $params = \Drupal::routeMatch()->getParameters();
+
+  $node_full_view_routes = [
+    'entity.node.canonical',
+    'entity.node.revision',
+    'entity.node.preview',
+  ];
+
+  if (in_array(\Drupal::routeMatch()->getRouteName(), $node_full_view_routes)) {
+    return $params->get('node');
+  }
+  else {
+    return FALSE;
+  }
```

Comment:
Nit, it's more typical for a function like this to return `NULL` rather than a boolean `FALSE`, in which case you could omit the whole `else` condition here.